src/python/pants/core/util_rules/search_paths.py

```diff
@@ -22,6 +28,74 @@ class ValidateSearchPathsRequest:
     local_only: FrozenOrderedSet[str]
 
 
+@dataclass(frozen=True)
+class VersionManagerSearchPathsRequest:
+    env_tgt: EnvironmentTarget
+    root_dir: str | None
+    tool_path: str
+    option: str
+    version_files: tuple[str, ...] = tuple()
+    local_token: str | None = None
+
+
+class VersionManagerSearchPaths(DeduplicatedCollection[str]):
+    pass
+
+
+@_uncacheable_rule
+async def get_un_cachable_version_manager_paths(
+    request: VersionManagerSearchPathsRequest,
+) -> VersionManagerSearchPaths:
+    """Inspects the directory of a version manager tool to find installations."""
```

Comment:
Might be good to mention a few version managers by name here to assist with grepping in future.